pkg/api/meta/restmapper.go

```diff
@@ -148,6 +148,23 @@ func kindToResource(kind string, mixedCase bool) (plural, singular string) {
 	return
 }
 
+// ResourceSingularlizer converts a resource name from plural to singular (e.g., from pods to pod)
+func ResourceSingularlizer(resource string) (singular string) {
+	if len(resource) == 0 {
+		return
+	}
+	if strings.HasSuffix(resource, "ies") {
```

Comment:
The singularizer was changed to a map (map between singular and plural), so we don't need this anymore.